apps/src/templates/ReCaptchaDialog.jsx

```diff
@@ -0,0 +1,109 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import i18n from '@cdo/locale';
+import Button from './Button';
+import BaseDialog from './BaseDialog';
+import DialogFooter from './teacherDashboard/DialogFooter';
+import Spinner from '../code-studio/pd/components/spinner';
+
+const styles = {
+  dialog: {
+    paddingLeft: 20,
+    paddingRight: 20,
+    paddingBottom: 20
+  }
+};
+
+export default class ReCaptchaValidationDialog extends React.Component {
+  static propTypes = {
+    handleSubmit: PropTypes.func.isRequired,
+    handleCancel: PropTypes.func.isRequired,
+    isOpen: PropTypes.bool.isRequired,
+    submitText: PropTypes.string.isRequired
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      disableSubmitButton: true,
+      loadedCaptcha: false
+    };
+    this.token = '';
+    this.onCaptchaVerification = this.onCaptchaVerification.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.onCaptchaExpiration = this.onCaptchaExpiration.bind(this);
+  }
+
+  componentDidMount() {
+    //Add reCaptcha and associated callbacks.
+    const script = document.createElement('script');
+    script.src = 'https://www.google.com/recaptcha/api.js';
+    script.id = 'captcha';
+    window.onSubmit = token => this.onCaptchaVerification(token);
```

Comment:
Maybe name this `onCaptchaSubmit` since it's in the global namespace?  (Also, consider renaming `captchaExpired` below to `onCaptchaExpired` for symmetry?)

Reply:
Good point - best to be as specific as possible in this situation.

Reply:
Thank you!